extensions/systemd/tests/ConsumeJournaldTest.cpp

```diff
@@ -90,7 +90,7 @@ struct TestJournal final : libwrapper::Journal {
   }
 
   int getCursor(gsl::owner<char*>* const cursor_out) noexcept override {
-    *cursor_out = gsl::owner<char*>(strdup(std::to_string(consumed).c_str()));
+    *cursor_out = strdup(std::to_string(consumed).c_str());  // NOLINT(cppcoreguidelines-owning-memory)
```

Comment:
Why remove the `gsl::owner` annotation?

Reply:
As the `cursor_out` parameter is already a `gsl::owner` type the cast and according to [these examples](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/owning-memory.html) that should be enough to mark the returned pointer's onwership. The previous annotation was marked as a C-style cast, but I think the current `cppcoreguidelines-owning-memory` warning is also a false positive so it was suppressed.

Reply:
There was no example that assigned owner through a pointer parameter. The examples don't have an additional layer of indirection over the owner.

Could you double check that leaving the gsl::owner annotation in place doesn't solve the warning? The C-style cast one should probably be just disabled, due to the high number of false positives.

Reply:
Reverted in b1d4a35c9a2021f2bd8d0fa750e2f2971f635cae